Project search bar layout improvements

alygin opened the pull request:
This PR matches project search layout with the recent changes in the buffer project layout.

No functional changes, just moving and adjusting visual elements.

Release Notes:

- Improve project search bar layout, match it with the buffer search bar.

Related Issues:

- #7722 

How it looks:

https://github.com/zed-industries/zed/assets/2101250/91b905ea-aed8-4740-9e60-67f3052885e2

alygin committed:
Project search bar layout improvements

SomeoneToIgnore reviewed (commented):
Thank you.

* One thing I've found very confusing is the `tab` action behavior when all inputs are open:

![image](https://github.com/zed-industries/zed/assets/2690773/ba7bc450-d10a-4338-aa0a-d06755d374a5)

if I start in the search field and tab my way through, I expect the caret to jump to each element in order of appearance, but now it comes to `replace` one last.
Previous version had the same flaw, but there it's not that visible due to different element positions.

* That `1/5` thing looks too distant from `<>` controls. I think, current version does not bother showing "No Matches" and has no issues due to that, should we do the same?

* Optional: while it's not the issue brought by current PR, should we slightly increase the height of `Search` and `Include` inputs? 
When I toggle between Regex and Semantic options, icons in those inputs disappear, making the inputs to change their height slightly which looks sad.

alygin committed:
Fix minor project search bar layout issues

alygin commented:
@SomeoneToIgnore,

> One thing I've found very confusing is the tab action behavior when all inputs are open:

Fixed

> That 1/5 thing looks too distant from <> controls. I think, current version does not bother showing "No Matches" and has no issues due to that, should we do the same?

If we don't show "No matches", we'll need to also hide <> buttons, or push them to the left, otherwise there will be a wide empty space. So' we'll have moving buttons, which is a controversial decision. Other editors try to avoid it for a reason:

<img width="553" alt="intellij-find-bar" src="https://github.com/zed-industries/zed/assets/2101250/2922cb93-7826-4149-9b69-499e875b3a57">
<img width="424" alt="vscode-find-bar" src="https://github.com/zed-industries/zed/assets/2101250/a8b5c700-f46d-479c-8455-7e1602a2702a">

Actually, I also think this layout is not perfect, but I think it's better than other options. Though, centering the text between the "Toggle replace" button and nav buttons (as JetBrains does, see above) may be a good alternative. What do you think?

BTW, the current project search panel doesn't bother to show "No matches", yet displays the "No results" message in the results area when the user searches for something and then erases the query.

> should we slightly increase the height of Search and Include inputs?

Sure, fixed.

SomeoneToIgnore reviewed (approved):
Sure, totally agree that moving <> around causes even more blinking and confusion.

Thank you for the fixes, looks much better than the current one!
